### PR TITLE
UI Configurable Page Sizes

### LINKED ISSUE #3523: [ui] [improvement] Configurable page-size for Jobs and Clients
### Nomad version
0.7.0

### Issue
The Job UI defaults pagination to 10 items. It would be ideal to make this configurable (i.e. the typical: display 10, 25, 50, 100) size drop down. This is also similar for the client workflow (top 8).

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
Perhaps the most requested feature? Configurable page sizes for Jobs, Clients, Allocations, and Volumes. 

When you select a page size preference, it is persisted in `localStorage` and applies to all paginated lists.

image

Closes #3523

### CODE REVIEW

File: ui/tests/acceptance/clients-list-test.js [hunk 1]
```diff
@@ -174,6 +178,17 @@ module('Acceptance | clients list', function(hooks) {
     assert.equal(currentURL(), '/settings/tokens');
   });
 
+  pageSizeSelect({
```
Comment (meirish) [inline ui/tests/acceptance/clients-list-test.js:181]: 👏 really like this pattern

File: ui/tests/pages/jobs/list.js [hunk 2]
```diff
@@ -64,6 +64,17 @@ export default create({
...
+    isPresent: isPresent('[data-test-page-size-select]'),
+    open: clickable('[data-test-page-size-select] .ember-power-select-trigger'),
+    selectedOption: text('[data-test-page-size-select] .ember-power-select-selected-item'),
+    options: collection('.ember-power-select-option', {
+      testContainer: '#ember-testing',
+      resetScope: true,
+      label: text(),
+    }),
+  },
+
```
Comment (backspace) [inline ui/tests/pages/jobs/list.js:77]: This looks like the third repetition of the identical page object component, maybe worthy of extraction?

File: ui/app/services/user-settings.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import Service from '@ember/service';
+import localStorageProperty from 'nomad-ui/utils/properties/local-storage';
+
+export default Service.extend({
+  pageSize: localStorageProperty('nomadPageSize', 25),
+});
```
Comment (backspace) [inline ui/app/services/user-settings.js:6]: I appreciate this facility for centralising user settings and how it provides a nice abstraction for the possible wonderful future where user data can magically persist outside a particular client 🤩🤞

File: ui/app/templates/jobs/index.hbs [hunk 4]
```diff
@@ -93,15 +93,16 @@
           {{/t.body}}
         {{/list-table}}
         <div class="table-foot">
+          {{page-size-select onChange=(action resetPagination)}}
```
[the remark this answers is not in the document]
Comment (DingoEatingFuzz) [reply ui/app/templates/jobs/index.hbs:96]: Wow, thanks for QAing for me, I should have caught this 🤦‍♂ 

I think there is at the very least a better way to handle this. I'll give it some time today.

File: ui/tests/pages/jobs/list.js (hunk 2 above)
Comment (DingoEatingFuzz) [reply ui/tests/pages/jobs/list.js:77]: I agree.

File: ui/app/controllers/csi/volumes/index.js [hunk 5]
```diff
@@ -24,12 +25,19 @@ export default Controller.extend(
...
 
     sortProperty: 'id',
     sortDescending: true,
 
     listToSort: alias('model'),
     sortedVolumes: alias('listSorted'),
+
+    // TODO: Remove once this page gets search capability
+    resetPagination() {
+      if (this.currentPage != null) {
```
[the remark this answers is not in the document]
Comment (DingoEatingFuzz) [reply ui/app/controllers/csi/volumes/index.js:38]: This is a "safe" thing to do that will compare against only `null` and `undefined` given JavaScript's truthiness rules, but it's admittedly clever and comes down to style.

There's an eslint rule that describes the idea (aptly named "smart"): https://eslint.org/docs/rules/eqeqeq#smart

[the remark this answers is not in the document]
Comment (backspace) [reply ui/app/controllers/csi/volumes/index.js:38]: ah news to me! Keeping it this way doesn’t bother me.

File: ui/tests/pages/jobs/list.js (hunk 2 above)
Comment (DingoEatingFuzz) [reply ui/tests/pages/jobs/list.js:69]: Officially extracted.

File: ui/app/templates/jobs/index.hbs (hunk 4 above)
[the remark this answers is not in the document]
Comment (DingoEatingFuzz) [reply ui/app/templates/jobs/index.hbs:96]: Fixed this.
Screen Shot 2020-04-02 at 1 40 13 PM

**Pull request merged**